daemons/dss-scalability-test/app.py

```diff
@@ -0,0 +1,335 @@
+import datetime
+import json
+import logging
+import os
+import sys
+import tempfile
+import time
+from decimal import Decimal
+
+import boto3
+import domovoi
+
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), 'domovoilib'))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+from dss import stepfunctions, Config, BucketConfig
+from dss.stepfunctions import generator
+from hca.dss import DSSClient
+from dss.api.files import ASYNC_COPY_THRESHOLD
+from json_generator import generate_sample
+
+#: Wait in seconds begore performing another checkout readiness check
+WAIT_CHECKOUT = 10
+
+#: Number of parallel execution branches within the scale test step function
+PARALLELIZATION_FACTOR = 10
+
+app = domovoi.Domovoi()
+
+app.log.setLevel(logging.DEBUG)
+
+test_bucket = os.environ["DSS_S3_CHECKOUT_BUCKET"]
+
+os.environ["HOME"] = "/tmp"
+
+os.environ["HCA_CONFIG_FILE"] = "/tmp/config.json"
+with open(os.environ["HCA_CONFIG_FILE"], "w") as fh:
+    fh.write(json.dumps({"DSSClient": {"swagger_url": "https://dss.dev.data.humancellatlas.org/v1/swagger.json"}}))
+
+client = DSSClient()
+
+dynamodb = boto3.resource('dynamodb')
+Config.set_config(BucketConfig.NORMAL)
+
+
+def current_time():
+    return int(round(time.time() * 1000))
+
+def upload_bundle(event, context, branch_id):
+    app.log.info("Upload bundle")
+    with tempfile.TemporaryDirectory() as src_dir:
+        with tempfile.NamedTemporaryFile(dir=src_dir, suffix=".json", delete=False) as jfh:
+            jfh.write(bytes(generate_sample(), 'UTF-8'))
+            jfh.flush()
+        with tempfile.NamedTemporaryFile(dir=src_dir, suffix=".bin") as fh:
+            fh.write(os.urandom(ASYNC_COPY_THRESHOLD + 1))
+            fh.flush()
+            start_time = current_time()
+            bundle_output = client.upload(src_dir=src_dir, replica="aws", staging_bucket=test_bucket)
+            return {"bundle_id": bundle_output['bundle_uuid'], "start_time": start_time}
+
+
+def download_bundle(event, context, branch_id):
+    app.log.info("Download bundle")
+    bundle_id = event['bundle']['bundle_id']
+    with tempfile.TemporaryDirectory() as dest_dir:
+        client.download(bundle_id, replica="aws", dest_name=dest_dir)
+    return {}
+
+
+def checkout_bundle(event, context, branch_id):
+    bundle_id = event['bundle']['bundle_id']
+    app.log.info(f"Checkout bundle: {bundle_id}")
+    checkout_output = client.post_bundles_checkout(uuid=bundle_id, replica='aws', email='rkisin@chanzuckerberg.com')
```

Comment:
I wouldn't recommend putting your email in here since this is public

Reply:
fixed